nifi-toolkit/nifi-toolkit-tls/src/main/java/org/apache/nifi/toolkit/tls/standalone/TlsToolkitStandalone.java

```diff
@@ -0,0 +1,101 @@
+import org.apache.nifi.toolkit.tls.properties.NiFiPropertiesWriterFactory;
+import org.apache.nifi.toolkit.tls.util.OutputStreamFactory;
+import org.apache.nifi.toolkit.tls.util.TlsHelper;
+import org.bouncycastle.openssl.jcajce.JcaMiscPEMGenerator;
+import org.bouncycastle.util.io.pem.PemWriter;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.security.KeyStore;
+import java.security.cert.X509Certificate;
+import java.util.List;
+
+public class TlsToolkitStandalone {
+    public static final String NIFI_KEY = "nifi-key";
+    public static final String NIFI_CERT = "nifi-cert";
+    public static final String ROOT_CERT_PRIVATE_KEY = "rootCert.key";
+    public static final String ROOT_CERT_CRT = "rootCert.crt";
+    public static final String NIFI_PROPERTIES = "nifi.properties";
+
+    private final OutputStreamFactory outputStreamFactory;
+
+    public TlsToolkitStandalone() {
+        this(FileOutputStream::new);
+    }
+
+    public TlsToolkitStandalone(OutputStreamFactory outputStreamFactory) {
+        this.outputStreamFactory = outputStreamFactory;
+    }
+
+    public void createNifiKeystoresAndTrustStores(File baseDir, TlsConfig tlsConfig, NiFiPropertiesWriterFactory niFiPropertiesWriterFactory, List<String> hostnames, List<String> keyStorePasswords,
+                                                  List<String> keyPasswords, List<String> trustStorePasswords, String httpsPort) throws GeneralSecurityException, IOException {
+        String signingAlgorithm = tlsConfig.getSigningAlgorithm();
+        int days = tlsConfig.getDays();
+        String keyPairAlgorithm = tlsConfig.getKeyPairAlgorithm();
+        int keySize = tlsConfig.getKeySize();
+        TlsCertificateAuthorityManager tlsCertificateAuthorityManager = new TlsCertificateAuthorityManager(tlsConfig);
+        KeyStore.PrivateKeyEntry privateKeyEntry = tlsCertificateAuthorityManager.getOrGenerateCertificateAuthority();
+        X509Certificate certificate = (X509Certificate) privateKeyEntry.getCertificateChain()[0];
+        KeyPair caKeyPair = new KeyPair(certificate.getPublicKey(), privateKeyEntry.getPrivateKey());
+
+        try (PemWriter pemWriter = new PemWriter(new OutputStreamWriter(outputStreamFactory.create(new File(baseDir, ROOT_CERT_CRT))))) {
+            pemWriter.writeObject(new JcaMiscPEMGenerator(certificate));
+        }
+
+        try (PemWriter pemWriter = new PemWriter(new OutputStreamWriter(outputStreamFactory.create(new File(baseDir, ROOT_CERT_PRIVATE_KEY))))) {
+            pemWriter.writeObject(new JcaMiscPEMGenerator(caKeyPair));
+        }
+
+        for (int i = 0; i < hostnames.size(); i++) {
+            String hostname = hostnames.get(i);
+            File hostDir = new File(baseDir, hostname);
+
+            if (!hostDir.mkdirs()) {
```

Comment:
thinking it makes sense to bomb earlier if the directory already exists and isn't empty to avoid having to deal with lots of edge cases